### ISSUE TITLE
Find references for Go in Sourcegraph.com returns incorrect results

### ISSUE THREAD

**Issue by nicksnyder** [sourcegraph/sourcegraph#5575]
Reproduction steps:
1. Go to https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/frontend/graphqlbackend/search_results.go#L856:37
2. Click "Go to definition" on `searchRepositories`
3. Click "Find references" on `searchRepositories`

Expected:
References for `searchRepositories`

Actual:
References for "github"

video: https://cl.ly/e31d0f482eef

screenshot:
![image](https://user-images.githubusercontent.com/754768/64886678-97c14e80-d61b-11e9-9c9a-1c61d2bb886b.png)

My user config:
```
{
  "typescript.accessToken": "redacted",
  "go.accessToken": "redacted",
  "git.blame.lineDecorations": true,
  "extensions": {
    "sourcegraph/code-discussions": false,
    "sourcegraph/codecov": true
  },
  "codecov.decorations.lineCoverage": false,
  "python.accessToken": "redacted"
}
```

**Comment by chrismwendt** [comment]
I'm able to repro, and this looks like a bug somewhere in the Sourcegraph extension API implementation.

Here's what I found: basic-code-intel's references provider is being called with the correct `doc.uri`, but incorrect `doc.text`. The given `doc.text` is for the previous document, not the current one. That's causing basic-code-intel to grab the wrong token to search for.

How I found that: apply this diff to basic-code-intel, run `./dev`, sideload it, disable `sourcegraph/go`, and refresh the page

```diff
diff --git a/package/src/handler.ts b/package/src/handler.ts
index a66fb16..f0d1b4c 100644
--- a/package/src/handler.ts
+++ b/package/src/handler.ts
@@ -764,6 +764,7 @@ export class Handler {
         if (doc.text === undefined) {
             return null
         }
+        console.log('refs', doc.uri, doc.text.slice(0, 500), pos)
         const tokenResult = findSearchToken({
             text: doc.text,
             position: pos,
```

First do the j2d, then refs. Check the console:

![image](https://user-images.githubusercontent.com/1387653/64895398-b33b5200-d639-11e9-9bfb-99207cc16979.png)

That's the text of `search_results.go`, not the current file which is `search_repositories.go`.

This looks similar to https://github.com/sourcegraph/sourcegraph/issues/5233
